src/features/streamlabs/http-server.spec.ts

```diff
@@ -0,0 +1,61 @@
+import http from "node:http";
+
+import { startHttpServer } from "./http-server";
+import { PortConflictError } from "./errors/port-conflict-error";
+import { PortPermissionError } from "./errors/port-permission-error";
+import { NetErrorCode } from "./errors/net-error";
+
+describe("http-server", () => {
+  test("http server starts on an unused port", async () => {
+    const server = await startHttpServer(1234);
+    expect(server).toBeInstanceOf(http.Server);
+
+    // Cleanup
+    await new Promise((resolve) => server!.close(resolve));
+    expect(server!.listening).toBe(false);
+  });
+
+  test("detects conflicts when ports are in use", async () => {
+    const port = 1234;
+
+    const server = await startHttpServer(port);
+    expect(server).toBeInstanceOf(http.Server);
+
+    // Starting a second server on the same port
+    await expect(startHttpServer(port)).rejects.toThrow(PortConflictError);
+
+    // Cleanup
+    await new Promise((resolve) => server!.close(resolve));
+    expect(server!.listening).toBe(false);
+  });
+
+  // Ports below 1024 require elevated permissions on some Operating Systems
```

Comment:
TIL

Reply:
If you ran these tests on Windows without the mocking, this test would fail since the server would succesfully open the port and the function would not throw. Also, jest would complain about an open handle (the TCP connection waiting)